src/hardware/disk_noise.cpp

```diff
@@ -0,0 +1,456 @@
+
+	auto* str_prop = secprop.Add_string("floppy_spin_up_sample",
+	                                    only_at_start,
+	                                    "fdd_spinup.wav");
+	assert(str_prop);
+	str_prop->Set_help(
+	        "Path to a .wav file for the floppy spin-up noise.\n"
+	        "Leave empty to not play this noise.\n");
+
+	str_prop = secprop.Add_string("floppy_spin_sample", only_at_start, "fdd_spin.wav");
+	assert(str_prop);
+	str_prop->Set_help(
+	        "Path to a .wav file for the floppy spin noise emulation.\n"
+	        "Leave empty to not play this noise.\n");
+
+	auto* int_prop = secprop.Add_int("floppy_spin_volume", only_at_start, 20);
+	int_prop->SetMinMax(0, 100);
+	int_prop->Set_help(
+	        "Set the volume of the floppy disk spin sound between 0 to 100 (Default: 20).\n");
+
+	int_prop = secprop.Add_int("floppy_seek_volume", only_at_start, 60);
+	int_prop->SetMinMax(0, 100);
+	int_prop->Set_help(
+	        "Set the volume of the floppy disk seek sound between 0 to 100 (Default: 60).\n");
+
+	str_prop = secprop.Add_string("hdd_spin_up_sample",
+	                              only_at_start,
+	                              "hdd_spinup.wav");
+	assert(str_prop);
+	str_prop->Set_help(
+	        "Path to a .wav file for the hard disk spin-up noise.\n"
+	        "Leave empty to not play this noise.\n");
+
+	str_prop = secprop.Add_string("hdd_spin_sample", only_at_start, "hdd_spin.wav");
+	assert(str_prop);
+	str_prop->Set_help(
+	        "Path to a .wav file for the hard disk spin noise emulation.\n"
+	        "Leave empty to not play this noise.\n");
+
+	int_prop = secprop.Add_int("hdd_spin_volume", only_at_start, 40);
+	int_prop->SetMinMax(0, 100);
+	int_prop->Set_help(
+	        "Set the volume of hard disk spin sound between 0 to 100 (Default: 10).\n");
+
+	int_prop = secprop.Add_int("hdd_seek_volume", only_at_start, 20);
+	int_prop->SetMinMax(0, 100);
+	int_prop->Set_help(
+	        "Set the volume of the hard disk seek sound between 0 to 100 (Default: 40).\n");
+
+	for (int i = 1; i <= 9; ++i) {
+		const std::string idx = std::to_string(i);
+
+		auto* fseek_prop =
+		        secprop.Add_string("floppy_seek_sample_" + idx,
+		                           Property::Changeable::OnlyAtStart,
+		                           ("fdd_seek" + idx + ".wav").c_str());
+		assert(fseek_prop);
+		fseek_prop->Set_help("Path to floppy seek sample " + idx + ".");
+
+		auto* hseek_prop =
+		        secprop.Add_string("hdd_seek_sample_" + idx,
+		                           Property::Changeable::OnlyAtStart,
+		                           ("hdd_seek" + idx + ".wav").c_str());
+		assert(hseek_prop);
+		hseek_prop->Set_help("Path to hard disk seek sample " + idx + ".");
+	}
+}
+
+void DISKNOISE_AddConfigSection(const ConfigPtr& conf)
+{
+	assert(conf);
+
+	constexpr auto changeable_at_runtime = false; // TODO: Deal with runtime
+	                                              // changes
```

Comment:
Enabling/disabling this at runtime is all we need, yeah. So the loading of sample would be done lazily, and that's about it.